src/zalopay/typings/agreementBindRequest.ts

```diff
@@ -0,0 +1,135 @@
+/*
+* The version of the ZaloPay OpenAPI document: v1.0.0
+* Contact: developer@zalopay.vn
+*
+* NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
+* https://openapi-generator.tech
+* Do not edit this class manually.
+*/
+
+export class AgreementBindRequest {
+        /**
+        * The unique ID of the application will be provided after the merchant registered successfully with ZaloPay.
+        */
+        'app_id': number;
+
+        /**
+        * Unique transaction ID of application, format: yyMMddxxxxxxxxx. Max length is 40. For example: 180208181007242
+        */
+        'app_trans_id': string;
+
+        /**
+        * The timestamp that order was created at in ms. The different limit is 15 minutes.
+        */
+        'req_date': number;
+
+        /**
+        * Maximum permissible amount. The default is real balance.
+        */
+        'max_amount'?: number;
+
+        /**
+        * \"URL of merchant binding page(web), be used in the desktop binding scenario. After the binding process finished, ZaloPay auto-debit binding page will redirect to merchant binding page via redirect URL with the binding result.\"
+        */
+        'redirect_url'?: string;
+
+        /**
+        * \"The deep-link of merchant app, be used in the mobile binding  scenario. After the binding process finished, ZaloPay app  will open the merchant app via redirect deep link with the  binding result.\"
+        */
+        'redirect_deep_link'?: string;
+
+        /**
+        * JSON string contain binding information, describe which information user will agree to allow merchant to do, using confirmation binding.
+        */
+        'binding_data'?: string;
+
+        /**
+        * This field specify which type of binding, currently only WALLET is allowed. Default is WALLET if not provided
+        */
+        'binding_type'?: AgreementBindRequest.BindingTypeEnum;
+
+        /**
+        * TBD
+        */
+        'callback_url'?: string;
+
+        /**
+        * User\'s identifier on merchant system, it can be merchant user\'s id, phone number, email...
+        */
+        'identifier': string;
+
+        /**
+        * It is signature of request. It\'s calculated by following input: hmacinput = app_id + `|` + apps_trans_id + `|` + binding_data + `|` + binding_type + `|` + identifier + `|` + max_amount + `|` + req_date; and use sha256 with app\'s hmac key as sign key.
+        */
+        'mac': string;
```

Comment:
@noradomi

Reply:
@leonthaibao I generated models as classes, not interface. So, we can use it with `new` and don't need worry about required fields.  
<img width="645" alt="image" src="https://user-images.githubusercontent.com/38036797/236724335-17414d90-2450-463e-928f-56e003942516.png">

Reply:
I got it, but I wonder if we can adjust the template to support that if the field is `app_id` or `mac`, we will make it optional. 
If we can do this, we can make a better experience to create an object in TS (in my opinion)

Reply:
> Can you share about it (better experience) between using interface vs class?

It's not about class or interface. You can use the class to verify the object just like the interface (the image that I captured)
The issue is TS cannot warn you which required field is missed with the current approach.

Reply:
@leonthaibao yeah, I understand. 

I've changed `app_id`, `mac` to optional fields by manual because mustache is not support for it (mustache is logic-less template). 
For new APIs followed our standard, generated model will be strong consistent with the specification without these cheats. So, I think this approach still be appliable.

![image](https://user-images.githubusercontent.com/38036797/236982006-fa889c6f-f069-4f07-8cab-e64df19beefb.png)

Reply:
@leonthaibao any should be updated?

Reply:
@noradomi No, I agree with you.